test/unit/plugin/v2/loader_test.rb

```diff
@@ -244,4 +248,82 @@ def test_when_a_train_plugin_is_installed_via_gem_and_required
     assert_includes(Train::Plugins.registry.keys, "test-fixture", "After requiring the gem, the Train Registry should know the plugin is loaded")
     assert(reg.loaded_plugin?(plugin_name), "After requiring, InSpec Registry should know the the plugin is loaded")
   end
+
+  REG_INST = Inspec::Plugin::V2::Registry.instance
+
+  def with_empty_registry
+    old_reg = REG_INST.registry.dup
+    REG_INST.registry.clear
+
+    yield
+  ensure
+    REG_INST.registry.replace old_reg
+  end
+
+  def with_logger
+    old_logger = Inspec::Log.logger
+    io = StringIO.new
+    Inspec::Log.logger = Logger.new(io)
+
+    yield
+
+    io.string
+  ensure
+    Inspec::Log.logger = old_logger
+  end
+
+  def using_bundler?
+    Gem::Specification.find_by_name("inspec")
+  rescue Gem::MissingSpecError
+    nil
+  end
+
+  def assert_detect_system_plugins(exp_keys, exp_err)
+    # rubocop:disable Style/HashSyntax
+    loader = Inspec::Plugin::V2::Loader.new(:omit_user_plugins => true,
+                                            :omit_bundles      => true,
+                                            :omit_core_plugins => true,
+                                            :omit_sys_plugins  => true)
+
+    assert_empty REG_INST.registry
+
+    yield loader
+
+    log_out = with_logger do
+      assert_silent do
+        loader.detect_system_plugins
+      end
+    end
+
+    assert_equal exp_keys, REG_INST.registry.keys.sort
+    assert_match exp_err, log_out
+  end
+
+  def test_detect_system_plugins
+    with_empty_registry do
+      exp = []
+      exp_err = "inspec gem not found, skipping detecting of system plugins\n"
+
+      assert_detect_system_plugins exp, exp_err do |loader|
+        def loader.find_inspec_gemspec(*)
```

Comment:
This feels like re-inventing mocking. Is there a reason the implementation of Inspec::Plugin::V2::Loader.find_inspec_gemspec can't be mocked?

Reply:
This is stubbing, the ruby way. Not re-invented, since this has existed in ruby since before mocking frameworks existed (literally).